templates/statefulset.yaml

```diff
@@ -0,0 +1,151 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "infinispan-helm-charts.name" . }}
+  labels:
+  {{- include "infinispan-helm-charts.labels" . | nindent 4 }}
+  annotations:
+    "openshift.io/display-name": "Infinispan Cluster"
+    "openshift.io/documentation-url": "http://infinispan.org/documentation/"
+spec:
+  serviceName: ""
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: infinispan-pod
+      clusterName: {{ include "infinispan-helm-charts.name" . }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- if not (.Values.security.endpointSecretName) }}
+        checksum/identities: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+      {{- end }}
+      labels:
+        app: infinispan-pod
+        clusterName: {{ include "infinispan-helm-charts.name" . }}
+    spec:
+      {{- if .Values.affinity }}
+      {{ .Values.affinity }}
+      {{- else }}
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    clusterName: {{ include "infinispan-helm-charts.name" . }}
+                    app: infinispan-pod
+                topologyKey: kubernetes.io/hostname
+              weight: 100
+      {{- end }}
+      {{if .Values.makeDataDirWritable }}
```

Comment:
Which envs this is required for again?

Reply:
Kubernetes still requires this option for both operator and helm chart.
Please see https://github.com/infinispan/infinispan-operator/issues/392

Reply:
From that issue it seems like Amazon EKS .. do you know if it happens on any other k8s distro?

Reply:
Sometimes it happens on different k8s distros in the cloud, don't remember but some upstream users asked about this option in the operator and we still have this flag there.